descqa/RedSequence.py

```diff
@@ -0,0 +1,226 @@
+        self.rs_z = fits.open(rs_path.format(self.validation_catalog, 'z'))[0].data[:-1]
+        self.rs_mean = fits.open(rs_path.format(self.validation_catalog, 'c'))[0].data[:-1]
+        self.rs_sigma = fits.open(rs_path.format(self.validation_catalog, 's'))[0].data[...,:-1]
+
+        self.z_bins = np.linspace(*kwargs.get('z_bins', (0.0, 1.0, 31)))
+
+        self.c_bins = np.linspace(*kwargs.get('c_bins', (-0.5, 2.0, 101)))
+        
+        self.mass_bins = np.logspace(*kwargs.get('mass_bins', (12.5, 14, 4)))
+
+        self.dz = self.z_bins[1:] - self.z_bins[:-1]
+        self.dz = np.hstack([self.dz, np.array([self.dz[-1]])])
+        self.dc = self.c_bins[1:] - self.c_bins[:-1]
+        self.dm = self.mass_bins[1:] - self.mass_bins[:-1]
+
+        self.n_z_bins = len(self.z_bins)-1
+        self.n_c_bins = len(self.c_bins)-1
+        self.n_mass_bins = len(self.mass_bins) - 1
+    
+        self.z_mean = (self.z_bins[1:] + self.z_bins[:-1]) / 2
+        self.c_mean = (self.c_bins[1:] + self.c_bins[:-1]) / 2
+        self.mass_mean = (self.mass_bins[1:] + self.mass_bins[:-1]) / 2
+        
+        self.use_redmapper = kwargs.get('use_redmapper', False)
+
+        possible_mag_fields = ('mag_true_{}_lsst',
+                               'mag_true_{}_des',
+                              'mag_true_{}_sdss',
+                              )
+        self.possible_mag_fields = [[f.format(self.bands[i]) for f in possible_mag_fields]
+                                    for i in range(len(self.bands))]
+
+        self.possible_absmag_fields = ('Mag_true_r_lsst_z0',
+                                       'Mag_true_r_lsst_z01'
+                                       'Mag_true_r_des_z0',
+                                       'Mag_true_r_des_z01',
+                                       'Mag_true_r_sdss_z0',
+                                       'Mag_true_r_sdss_z01',
+                                   )
+                               
+                               
+    def prepare_galaxy_catalog(self, gc):
+
+        quantities_needed = {'redshift_true', 'is_central', 'halo_mass', 'halo_id', 'galaxy_id'}
+
+        if gc.has_quantities(['truth/RHALO', 'truth/R200']):
+            gc.add_quantity_modifier('r_host', 'truth/RHALO', overwrite=True)
+            
+            gc.add_quantity_modifier('r_vir', 'truth/R200', overwrite=True)
+            quantities_needed.add('r_host')
+            quantities_needed.add('r_vir')
+
+        mag_fields = []
+        for i in range(len(self.bands)):
+            mag_fields.append(gc.first_available(*self.possible_mag_fields[i]))
+            quantities_needed.add(mag_fields[i])
+
+        absolute_magnitude_field = gc.first_available(*self.possible_absmag_fields)
+        quantities_needed.add(absolute_magnitude_field)
+
+        if ((not np.all(gc.has_quantities(mag_fields))) and gc.has_quantities(quantities_needed)):
```

Comment:
Since `mag_fields` are already added to  `quantities_needed`:
```suggestion
        if not gc.has_quantities(quantities_needed):
```